src/lib/compiler.ts

```diff
@@ -3862,20 +3867,20 @@ export default class Compiler {
       switch (v.type) {
         case 'global':
           if (isNumeric(v.valueType)) {
-            state.global.num_uints += 1;
+            state.global.num_uints += typeof v.maxKeys !== 'undefined' ? v.maxKeys : 1;
```

Comment:
This could be simplified with `v.maxKeys || 0`. We've seen some weird bugs when using `typeof` in algokit with some libraries.

Reply:
Should we use 1 or 0 for the default count?

Reply:
Ah yeah good point, maybe the default should 16 for local and 64 for global... that way if you have more than two maps with undefined max keys the contract will fail to be created. Or perhaps `maxKeys` is simply a required argument

Reply:
Required arguments would make sense, maybe we can do a check in `appSpec` and error out during compile if it is over allocated. Lets wrap back around tmrw and we can put the final polish on 🕺.